### PR TITLE
Add blog post "Mockify"

### PR DESCRIPTION
**Pull request by BirgitBader**
@ChrBrAtePages please check. I've just made some minor spelling adjustments and added a small introduction. Thanks.

### CODE REVIEW

File: _posts/2016/2016-02-04-mockify.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+Later, another developer decides to *rename* or *delete* the method that you mocked previously.
+If you run your tests now, all is fine.
+But outside of your test environment your code will **die**, because the method your code is calling does not exist anymore.
+It is unlikely to cover all this issues completely with integration tests.
+
+For this reason *Mockify* verifies that the method you are trying to mock is actually existing on the affected package.
+This was the starting point of *Mockify* and with the time I added more functionality to the framework.
+
+{% highlight ruby %}
+my $Mockify = DE_EPAGES::Test::Mock::Mockify->new('Package::Path::MyModule',['ParameterInConstructor']);
```
Comment (ubep) [inline _posts/2016/2016-02-04-mockify.md]: use path from https://github.com/ChrBrAtePages/Mockify

File: _posts/2016/2016-02-04-mockify.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+So you have to make sure that your mocked method was called with the correct amount of parameters.
+*Perl* also does not care about the *order of the parameters*.
+Your test has to ensure this, too.
+
+In *Mockify* you can define the *type*, and additionally, the *value* you expect for every parameter.
+
+{% highlight ruby %}
+$Mockify->mock('myMethodName','the return value',['string',{'int' => 123}]);
+{% endhighlight %}
+
```
Comment (ubep) [inline _posts/2016/2016-02-04-mockify.md]: A further sentence to explain the code example would be nice:
e.g.: "Here we expect two arguments, one arbitrary string and one integer which has to be exactly 123"

Comment (ubep) [inline _posts/2016/2016-02-04-mockify.md]: A further sentence to explain the code example would be nice:
e.g.: "Here we expect two arguments, one arbitrary string and one integer which has to be exactly 123"

File: _posts/2016/2016-02-04-mockify.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+### Collect meta data
+
+Sometimes the parameters can change at runtime, so you cannot check this with parameter checks at call-time.
+You have verify this at a later stage in the test.
```
Comment (ubep) [inline _posts/2016/2016-02-04-mockify.md]: have TO verify

File: _posts/2016/2016-02-04-mockify.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: "Mockify - A mocking framework for Perl"
+date: "2016-02-04 08:00:02"
+icon: recycle
+categories: tech-stories
+authors: ["Christian B."]
+---
+
+Already when reading the title of this post, you may ask yourself why hell do they deal with Perl?
```
Comment (ubep) [inline _posts/2016/2016-02-04-mockify.md]: why THE hell

File: _posts/2016/2016-02-04-mockify.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+## Why developing a mocking framework for Perl
+
+- Yes, *Perl* is a good old scripting Camel.
+- Yes, we already started the new version of ePages based on a modern software stack, but our core is written in *Perl* and we have to live with this for a while.
+- No, we cannot switch from one day to another to the new technology.
+- No, writing NO tests is not an option.
+
+We have to support a while the current core.
+Also we need to write unit-tests in our legacy code base.
```
Comment (ubep) [inline _posts/2016/2016-02-04-mockify.md]: unit-tests without dash

File: _posts/2016/2016-02-04-mockify.md (hunk 1 above)
Comment (ChrBrAtePages) [inline _posts/2016/2016-02-04-mockify.md]: my $Mockify = Mockify->new('Package::Path::MyModule',['ParameterInConstructor']);

**Pull request merged**